massa-signature/src/signature_impl.rs

```diff
@@ -20,6 +20,22 @@ const PRIVATE_KEY_STRING_PREFIX: &str = "PRI";
 const PUBLIC_KEY_STRING_PREFIX: &str = "PUB";
 const SIGNATURE_STRING_PREFIX: &str = "SIG";
 
+/// `KeyPair` is used for signature and decrypting
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct KeyPair {
```

Comment:
I think the shnorr crate has a KeyPair ! 
see the definition of our current PrivateKey, it already wraps a keypair

Reply:
It shouldn't add any constraints: the wrapped type should be private and only a couple methods should be exposed.

For example:

* struct PublicKey(lib::PubKey)
  * verify(data, signature) -> bool

* struct KeyPair(lib::KeyPair)
  * sign(data) -> Signature
  * get_public_key() -> &PublicKey

Reply:
I have implemented this interface.